### DISCUSSION TITLE
How to Backup sqlite database properly

### DISCUSSION THREAD

**Discussion by fdisamuel** [dani-garcia/vaultwarden#1613] [Q&A] [answered]
Hi!

I know this is a noob question, but I didn't find an exact answer in the wiki to this: 

When making backup of the sqlite database with the online Backup API (.backup), do I need to copy the `-shm`, and `-wal` files to my backup directory, or are these "handled" automatically? Furthermore upon restore is that enough if just copy my backup db.sqlite3 file (and all the other stuff except `-shm` and `-wal` files) to the Bitwarden data directory?

Thanks in advance.

**Comment by jjlin** [comment, marked as answer]
You only need to keep the single output file when using the `.backup` command. Restoring is as simple as what's stated at https://github.com/dani-garcia/bitwarden_rs/wiki/Backing-up-your-vault#restoring-backup-data.

**Reply by fdisamuel** [reply]
Thank you!
